scikeras/wrappers.py

```diff
@@ -871,192 +874,218 @@ def preprocess_y(y):
 
         Returns:
             y : modified 2D numpy array with 0 indexed integer class labels.
-            extra_args : dictionary of output attributes, ex `n_outputs_`
         """
-        y, extra_args = super(KerasClassifier, KerasClassifier).preprocess_y(y)
-
-        target_type_ = type_of_target(y)
+        y = super().preprocess_y(y, reset=reset)
 
-        if len(y.shape) == 1:
-            n_outputs_ = 1
-        else:
-            n_outputs_ = y.shape[1]
+        loss = self.loss
 
-        if target_type_ == "binary":
+        if self.target_type_ == "binary":
```

Comment:
It appears the purpose of this if-statement is to set these variables:

* `model_n_outputs_`, `n_outputs_`, `classes_`, `n_classes_`, `encoders_`, `n_classes_`.

Why not code these variables with dictionaries?

``` python
def preprocess_y(self, y):
    if target_type == "binary":
        meta = {
            "model_n_outputs_": ...,
            "n_outputs_": ...,
            ...
        }
    elif target_type == "multiclass":
        meta = {
            "model_n_outputs_": ...,
            "n_outputs_": ...,
            ...
        }
    for k, v in meta.items():
        setattr(self, k, v)
```

I'd imagine this refactor makes this function a lot simpler (easier to review, document and reason about).

Reply:
yeah we can do it that way!

Reply:
Incorporated.